src/Application/Game.cpp

```diff
@@ -1117,130 +1112,48 @@ void Game::EventLoop() {
                     DialogueEnding();
                     current_screen_type = CURRENT_SCREEN::SCREEN_GAME;
                     continue;
-                case UIMSG_CastSpell_Telekinesis:
-                    HEXRAYS_LOWORD(v42) = vis->get_picked_object_zbuf_val().object_pid;
-                    v44 = (uint16_t)v42;
-                    v45 = PID_TYPE(v44);
-                    uNumSeconds = v44;
-                    v46 = PID_ID(v44);
-                    if (v45 == OBJECT_Actor) {
-                        v47 = pActors[v46].uAIState == Dead;
-                        if (!v47) continue;
-                        pSpellInfo = static_cast<CastSpellInfo *>(pGUIWindow_CastTargetedSpell->wData.ptr);
-                        pSpellInfo->uFlags &= ~ON_CAST_Telekenesis;
-                        pSpellInfo->uPlayerID_2 = uMessageParam;
-                        pSpellInfo->spell_target_pid = v44;
-                        pParty->pPlayers[pSpellInfo->uPlayerID].SetRecoveryTime(
-                            300);
-                        pGUIWindow_CastTargetedSpell->Release();
-                        pGUIWindow_CastTargetedSpell = 0;
-                        mouse->SetCursorImage("MICON1");
-                        game_ui_status_bar_event_string_time_left = 0;
-                        IsEnchantingInProgress = false;
-                        back_to_game();
-                        continue;
+                case UIMSG_CastSpell_Telekinesis: {
+                    int pid = vis->get_picked_object_zbuf_val().object_pid;
+                    ObjectType type = PID_TYPE(pid);
+                    int id = PID_ID(pid);
+                    bool interactionPossible = false;
+                    if (type == OBJECT_Actor) {
+                        interactionPossible = pActors[id].uAIState == Dead;
+                    }
+                    if (type == OBJECT_Item) {
+                        int flags = pObjectList->pObjects[pSpriteObjects[id].uObjectDescID].uFlags;
+                        interactionPossible = (flags & OBJECT_DESC_UNPICKABLE) != OBJECT_DESC_UNPICKABLE;
```

Comment:
What I meant here is that you can just write:

`interactionPossible = !(flags & OBJECT_DESC_UNPICKABLE);`